bootstrap.js

```diff
@@ -4,101 +4,200 @@ Cu.import("resource://gre/modules/Console.jsm");
+    // Get the "DOM" elements
+    const urlBar = this.window.document.getElementById("urlbar");
+    // XUL elements are different than regular children
+    return this.window.document.getAnonymousElementByAttribute(urlBar, "anonid", "input");
+  }
+
+  get shareButton() {
+    return this.window.document.getElementById("social-share-button");
+  }
 
-  // Load the CSS with the shareButton animation
-  utils = bWindow.QueryInterface(Ci.nsIInterfaceRequestor).getInterface(Ci.nsIDOMWindowUtils);
-  utils.loadSheet(CSS_URI, utils.AGENT_SHEET);
+  insertCopyController() {
+    // refresh urlInput reference, this is potentially changed by the customize event
+    this.urlInput.controllers.insertControllerAt(0, this.copyController);
+  }
 
-  insertCopyController();
+  removeCopyController() {
+    // refresh urlInput reference, this is potentially changed by the customize event
+    this.urlInput.controllers.removeController(this.copyController);
+  }
+
+  animationEndListener(e) {
+    // When the animation is done, we want to remove the CSS class
+    // so that we can add the class again upon the next copy and
+    // replay the animation
+    this.shareButton.classList.remove("social-share-button-on");
+  }
+
+  addCustomizeListener() {
+    this.window.addEventListener("customizationending", this.insertCopyController);
+  }
+
+  removeCustomizeListener() {
+    this.window.removeEventListener("customizationending", this.insertCopyController);
+  }
+
+  insertCSS() {
+    const utils = this.window.QueryInterface(Ci.nsIInterfaceRequestor)
+      .getInterface(Ci.nsIDOMWindowUtils);
+    utils.loadSheet(CSS_URI, utils.AGENT_SHEET);
+  }
+
+  removeCSS() {
+    const utils = this.window.QueryInterface(Ci.nsIInterfaceRequestor)
+      .getInterface(Ci.nsIDOMWindowUtils);
+    utils.removeSheet(CSS_URI, utils.AGENT_SHEET);
+  }
+
+  // TODO Add startup and shutdown functions.
+  startup() {
+    browserWindowWeakMap.set(this.window, this);
+
+    // The customizationending event represents exiting the "Customize..." menu from the toolbar.
+    // We need to handle this event because after exiting the customization menu, the copy
+    // controller is removed and we can no longer detect text being copied from the URL bar.
+    // See DXR:browser/base/content/browser-customization.js
+    this.addCustomizeListener();
+
+    // Load the CSS with the shareButton animation
+    this.insertCSS();
+
+    // insert the copy controller to detect copying from URL bar
+    this.insertCopyController();
+  }
+
+  shutdown() {
+    // Remove the customizationending listener
+    this.removeCustomizeListener();
+
+    // Remove the CSS
+    this.removeCSS();
+
+    // Remove the copy controller
+    this.removeCopyController();
+
+    // Remove modifications to shareButton (modified in CopyController)
+    if (this.shareButton !== null) {
+      // if null this means the user did not copy from the URL bar
+      // so we don't have anything to remove
```

Comment:
I don't think this comment is correct anymore, since you changed to getters. I think that if `this.shareButton` is null, that means that there isn't a share button on the page.